### ISSUE TITLE
[App]: NaN values reported at incorrect times

### ISSUE THREAD

**Issue by maxime915** [wandb/wandb#6469]
### Current Behavior

When NaN values are encountered, the graph shows them as coming from before 1980 (see below).

In relative wall time, however, they is a large gap between the last non-NaN value and the first NaN. The last valid value is at 85 minutes and the first NaN at 5195.132 (a bit more than 86 hours while the whole run took less than 10 hours). The relative process time is similar.

There is no gap between the values when using steps as the x-axis.

### Expected Behavior

The time (both absolute and relative) of values that are NaN should be reported as those from the system.

### Steps To Reproduce

To be honest, I haven't isolated the issue in a simpler project. Here is what I did to observe it "live".

- Make an experiment that produces NaN values after some time
- Log these values
- Using the plot, change the axis to relative or absolute wall time

### Screenshots

Absolute wall time:
![Screenshot from 2023-10-18 07-50-59](https://github.com/wandb/wandb/assets/13822908/2ab67ed3-7f76-41fc-bb5d-eb867fabac93)

Relative wall time:
![Screenshot from 2023-10-18 07-52-02](https://github.com/wandb/wandb/assets/13822908/eb2992a0-c8d5-4832-a536-8b90ac3e2b5f)

Logging network issues: 
![Screenshot from 2023-10-18 07-55-58](https://github.com/wandb/wandb/assets/13822908/1e192071-f3dd-4b51-939e-2de527def716)

### Environment

OS: Linux-6.2.0-34-generic-x86_64-with-glibc2.35 (Ubuntu 22.04)

Browsers: Firefox

Version: 118.0.2 (64-bit)

### Additional Context

Additional info : there were some network issues (~1h before the first NaN), I have no idea whether they are related or not.

I must say, I am not certain that this issue comes from the WebApp or the Python client (0.15.8). Any help is appreciated. I tried to find a list of data that was sent to the server, but I couldn't find anything aside from images.

Here is how I log the values, both `train_1_epoch` and `val_log` call to `setup.logger.log` with `commit=False` (`setup.logger` is the return value of `wandb.init`). Note that the `"lr"` plot has no NaN values.

```python
def train(setup: Setup):
    if setup.config.warmup > 0:
        warmup(setup)

    for epoch in range(setup.config.num_epochs):
        train_1_epoch(setup, epoch)
        metrics = val_log(setup, epoch)
        if setup.scheduler is not None:
            setup.scheduler.step(metrics["val/loss_rec"])
        lr = float("nan")
        if len(setup.optimizer.param_groups) == 1:
            lr = setup.optimizer.param_groups[0]["lr"]
        # it would have been better to log it *before* the update to be relevant to the epoch
        setup.logger.log({"lr": lr}, commit=True)
        if should_stop(setup, epoch):
            break
```

**Comment by maxime915** [comment]
I'm managed to reproduce the issue for the absolute wall time, but not the relative wall time.
The project is public and accessible here https://wandb.ai/maxime915/issue-6469 . Here is the reproduction code

```python
import time

import wandb

wandb.init(project="issue-6469", entity="maxime915")

for epochs in range(10):
    # for batch, _ in enumerate(range(25)):
    #     ...

    if epochs > 6:
        epochs = float("nan")
    else:
        epochs = float(epochs)

    time.sleep(0.7)
    wandb.log({"train/loss": 0.89 ** epochs}, commit=False)

    time.sleep(0.3)
    wandb.log({"val/loss": 0.9 ** epochs}, commit=False)

    wandb.log({"lr": 0.1}, commit=True)

wandb.finish()
```

The absolute wall time is again completely off
![Screenshot from 2023-10-18 13-06-55](https://github.com/wandb/wandb/assets/13822908/4b1fc7c4-df11-415c-9da3-b7942cfeb719)

But the relative wall time is OK
![Screenshot from 2023-10-18 13-07-23](https://github.com/wandb/wandb/assets/13822908/bd14749c-d1ab-4a57-9e3b-4c14d883cf55)

Hard to say what caused the relative timestamp to be completely off in my original project but not this one.

**Comment by ArtsiomWB** [comment]
Hi @maxime915! Thank you for writing in and reporting this. 

Thank you so much for sending over the reproduction code, I was able to repro this on my side and will report it to our engineering team.
